tools/deploy-with-olm.sh

```diff
@@ -39,21 +39,47 @@ DEPLOYER_IMAGE=${DEPLOYER_IMAGE:-${IMAGE_REPO}/${DEPLOYER_NAME}:${DEPLOYER_VERSI
 # Deploy target
 TARGET_NAMESPACE=${TARGET_NAMESPACE:-openshift-storage}
 
+# CSV environment variables
+ADDON_NAME=${ADDON_NAME:-ocs-converged}
+SOP_ENDPOINT=${SOP_ENDPOINT:-"https://red-hat-storage.github.io/ocs-sop/sop/OSD/{{ .GroupLabels.alertname }}.html"}
+BUNDLE_FILE=bundle/manifests/ocs-osd-deployer.clusterserviceversion.yaml
+
+PD_KEY=${PD_KEY:-""}
+SNITCH_URL=${SNITCH_URL:-""}
+CLUSTER_SIZE=${CLUSTER_SIZE:-1}
+
 # Generate the deployer image 
 blue "Generate deployer image: ${DEPLOYER_IMAGE}"
 exit_on_err make docker-build IMG=${DEPLOYER_IMAGE}
 exit_on_err make docker-push IMG=${DEPLOYER_IMAGE}
 
 # Generate the olm bundle image
 blue "Generate and push the olm bundle image: ${BUNDLE_IMAGE}"
-exit_on_err make bundle IMG=${DEPLOYER_IMAGE}
+exit_on_err make bundle IMG=${DEPLOYER_IMAGE} bundle_dir=./tools/bundle
+exit_on_err sed -i "s|- name: ADDON_NAME|- {name: ADDON_NAME, value: ${ADDON_NAME}}|" ${BUNDLE_FILE}
+exit_on_err sed -i "s|- name: SOP_ENDPOINT|- {name: SOP_ENDPOINT, value: \'${SOP_ENDPOINT}\'}|" ${BUNDLE_FILE}
 exit_on_err make bundle-build BUNDLE_IMG=${BUNDLE_IMAGE}
 exit_on_err make docker-push IMG=${BUNDLE_IMAGE}
 
+# Install operator prereqs on cluster
+blue "Creating ${TARGET_NAMESPACE} on cluster"
+exit_on_err ${K8S_CLIENT} create namespace ${TARGET_NAMESPACE} --dry-run=client -o yaml | ${K8S_CLIENT} -n ${TARGET_NAMESPACE} apply -f -
+
+blue "Creating catalogsource containing needed dependencies on cluster"
+exit_on_err ${K8S_CLIENT} -n ${TARGET_NAMESPACE} apply -f ./tools/yamls/ocs-catalogsource.yaml
+
+blue "Creating secrets needed for the operator to run without error"
+exit_on_err ${K8S_CLIENT} -n ${TARGET_NAMESPACE} create secret generic addon-${ADDON_NAME}-parameters --from-literal=size=${CLUSTER_SIZE} --dry-run=client -o yaml | ${K8S_CLIENT} -n ${TARGET_NAMESPACE} apply -f -
+exit_on_err ${K8S_CLIENT} -n ${TARGET_NAMESPACE} create secret generic ${ADDON_NAME}-pagerduty --from-literal=PAGERDUTY_KEY=${PD_KEY} --dry-run=client -o yaml | ${K8S_CLIENT} -n ${TARGET_NAMESPACE} apply -f -
+exit_on_err ${K8S_CLIENT} -n ${TARGET_NAMESPACE} create secret generic ${ADDON_NAME}-deadmanssnitch --from-literal=SNITCH_URL=${SNITCH_URL} --dry-run=client -o yaml | ${K8S_CLIENT} -n ${TARGET_NAMESPACE} apply -f -
+
 # In install olm if needed
 blue "Checking for existing olm"
-${K8S_CLIENT} get namespace olm >/dev/null 2>&1
-if [ $? -eq 0 ]; then
+if ${K8S_CLIENT} get namespace olm >/dev/null 2>&1
+then
+    blue "olm found"
+elif ${K8S_CLIENT} get namespace openshift-operator-lifecycle-manager &> /dev/null
+then
```

Comment:
Please install OLM before deploying anything on the cluster. 
The cluster shouldn't be touched if OLM has failed to be deployed